subsys/net/lib/sockets/Kconfig

```diff
@@ -48,15 +48,16 @@ config NET_SOCKETS_DNS_TIMEOUT
 
 config NET_SOCKETS_SOCKOPT_TLS
 	bool "Enable TCP TLS socket option support [EXPERIMENTAL]"
-	select TLS_CREDENTIALS
+	imply TLS_CREDENTIALS if NET_NATIVE
+	imply MBEDTLS if NET_NATIVE
```

Comment:
Does `NET_SOCKETS_SOCKOPT` need `TLS_CREDENTIALS` and `MBEDTLS` to be enable if `NET_NATIVE` is enabled, or are they still optional?

If it needs them to be enabled, you could use `select`. Otherwise, `imply` is fine (which just turns them on by default).

Reply:
They are required, at least for now there is no other implementaion for TLS sockets. Will switch to `select`.